posthog/queries/person_query.py

```diff
@@ -197,21 +196,21 @@ def _get_grouped_person_filters(self, prepend: str = "") -> Tuple[str, Dict]:
             prepend=prepend,
             hogql_context=self._filter.hogql_context,
         )
-
-    def _get_person_filters(self, prepend: str = "") -> Tuple[str, Dict]:
-        return parse_prop_grouped_clauses(
+        prefiltering_conditions, prefiltering_params = parse_prop_grouped_clauses(
             self._team_id,
             self._inner_person_properties,
             has_person_id_joined=False,
             group_properties_joined=False,
+            # The above kwargs are the same as for finalization EXCEPT this one - we use the property
+            # from the person BEFORE aggregation by version here, to eliminate persons early on
             person_properties_mode=PersonPropertiesMode.DIRECT_ON_PERSONS,
             prepend=prepend,
```

Comment:
probably a pre-existing bug: if prepend is the same in the grouped & non-grouped version, the params generated would overwrite each other?

Reply:
welll, maybe for now, but given a different person property mode, they _could_ end up doing different things, which would then most definitely bork this

Reply:
Hmm, alright, disambiguated https://github.com/PostHog/posthog/commit/4860243e396308b8f8498c5367fcd4ece6391af6